spec/quotas/quotas.mdown

```diff
@@ -32,51 +35,95 @@ This document defines one additional capability URI.
 
+## The ResourceType Data Type
+
+The **ResourceType** is a `String` from an enumeration defined list of values, handled by the server.
+
+A resource type is like an unit of measure for the quota usage. Some custom specifications might provide some additional
+values. If the client does not specify custom resource type specifications in the "using" parameter of the request,
+the server should respond the JSON value `null`, instead of answering a resource type value that the client does not
+support. Standard values are:
+
+* `count`: The quota is measured in number of data type objects. For example, a quota can have a limit of 50 `Mail` objects.
+* `size`: The quota is measured in size. The default unit is in `bytes`, but a server can decide of the unit it wants
+to use (like in `octets`). For example, a quota can have a limit of 25000 `bytes`
+
 ## The Quota Object
 
 The quota object MUST contain the following fields:
 
 * **id**: `Id` The unique identifier for this object. It should respect the JMAP ID datatype defined in section 1.2 of [@!RFC8620]
 
-* **type**: `String` The type of quota. It should either be "messageCount" or "messageStorageSize". There is two ways we can define a quota: either on a number of emails, or in the amount of bytes.
+* **resourceType**: `ResourceType` The resource type of the quota.
 
 * **used**: `UnsignedInt` The current usage of the mailbox. Computation of this value is handled by the server.
 
-* **usedScope**: `Scope` The `Scope` of the current mailbox usage.
+* **limit**: `UnsignedInt` The hard limit set by this quota object. No more outgoing and ingoing messages should be
+allowed if we reach this limit. It should higher than the `warnLimit` and the `softLimit`.
+
+* **scope**: `Scope` The `Scope` of this quota.
 
-* **limit**: `UnsignedInt` The limit set by this quota object to the current mailbox.
+* **name**: `String` The name of the quota object. Useful for managing quotas and use queries for searching.
 
-* **limitScope**: `Scope` The `Scope` of the mailbox limit.
+* **datatypes**: `String[]` A list of all the data types values that are applying to this quota. This allows to assign
+quotas to separated or shared data types. This MAY include data types the client does not recognise. Clients MUST
+ignore any unknown data type in the list.
 
 The quota object MAY contain the following field:
 
-* **description**: `String|null` Arbitrary free, human readable, description of this quota. Might be used to explain where the limit comes from and explain the entities this quota applies to.
+* **warnLimit**: `UnsignedInt|null` The warn limit set by this quota object. It can be used to send a warning to an
+user that he is going to reach the hard limit soon, but no action is taken. If set, it should be lower than the
+`softLimit` and the `limit`.
+
+* **softLimit**: `UnsignedInt|null` The soft limit set by this quota object. It can be used to block outgoing messages,
```

Comment:
We need a non mail wording as it applies to the whole account. Some operations might be allowed, and some other defended?

Reply:
You are right, it doesn't concern only messages anymore

Reply:
Is it normal in these sorts of specifications to refer to the user as `he`?

Reply:
No, the RFCs should be gender neutral. I'm sure the RFC editors would pick this up before publication but better to fix this now.

Reply:
I should really get back to that spec... Thanks for the feedback guys, it is indeed a mistake !